app/services/foreman_openscap/lookup_key_overrider.rb

```diff
@@ -0,0 +1,71 @@
+module ForemanOpenscap
+  class LookupKeyOverrider
+    def initialize(policy)
+      @policy = policy
+      @name_service = ConfigNameService.new
+    end
+
+    def override
+      return unless @policy.deploy_by && Policy.deploy_by_variants.include?(@policy.deploy_by)
+      config = @name_service.config_for @policy.deploy_by.to_sym
+      unless config.available?
+        @policy.errors[:deploy_by] <<
+          _("%{type} was selected to deploy policy to clients, but %{type} is not available. Are you missing a plugin?") %
+          { :type => config.type.to_s.camelize }
+        return
+      end
+      return unless config.managed_overrides?
+      override_required_params config
```

Comment:
Interestingly, I was able to create policy with ansible deployment with only role imported, no variables were imported, I'd expect that also fails.

Reply:
with variables imported, it marked them for override but I didn't see the default value or matcher being set for port and server, it only set the erb for policies

Reply:
That is the same behaviour as we have for Puppet. There is no default value for port and server - they are based on smart proxy with openscap feature. They get populated once host(group) has openscap proxy.

Reply:
I don't think we'd force user to fill that in for smart class parameter manually. I tested that on Foreman 1.18 at least, after we create a policy, we add a matcher for hostgroup for both server and port parameter for this hostgroup. But that does not seem to happen for respective ansible variables. They are just set for overriding. It's fine not to set default value, but the matcher for hostgroup="theoneIselected" should be added. Sorry for confusing with default value.